### PR TITLE
Transform winston logs to kafka logger

### PR DESCRIPTION
**Pull request by guptad2**

### CODE REVIEW

File: config/toggles.js [hunk 1]
```diff
@@ -39,6 +39,23 @@ function environmentVariableTrue(processEnv, environmentVariableName) {
...
+/**
+ * Return boolean false if the named environment variable is boolean false or
+ * case-insensitive string 'false'.
+ *
+ * @param {Object} processEnv - The node process environment. (Passing it into
+ *  this function instead of just getting a reference to it *inside* this
+ *  function makes the function easier to test.)
+ * @param {String} environmentVariableName - The name of the environment var.
+ * @returns {Boolean} true if the named environment variable is boolean true or
+ *  case-insensitive string 'true'.
```
Comment (pallavi2209) [inline config/toggles.js]: true if the named environment variable is boolean **false**...

File: config/toggles.js [hunk 2]
```diff
@@ -39,6 +39,23 @@ function environmentVariableTrue(processEnv, environmentVariableName) {
     x.toString().toLowerCase() === 'true';
 } // environmentVariableTrue
 
+/**
+ * Return boolean false if the named environment variable is boolean false or
+ * case-insensitive string 'false'.
```
Comment (pallavi2209) [inline config/toggles.js]: Return boolean **true**...

File: config/toggles.js [hunk 3]
```diff
@@ -164,6 +181,12 @@ const longTermToggles = {
   enableRedirectDifferentInstance: environmentVariableTrue(pe,
     'ENABLE_REDIRECT_DIFFERENT_INSTANCE'),
 
+  // Toggle to turn on Kafka Logging
+  kafkaLogging: environmentVariableTrue(pe, 'KAFKA_LOGGING'),
+
+  // Toggle to turn on LocalLogging
```
Comment (pallavi2209) [inline config/toggles.js]: Please add a comment for the default behaviour of LocalLogging here.

File: tests/logger/logger.js [hunk 4]
```diff
@@ -11,9 +11,18 @@ const expect = require('chai').expect;
 const { initKafkaLoggingProducer, writeLog } = require('../../logger');
 const KafkaProducer = require('no-kafka');
 const sinon = require('sinon');
+const featureToggles = require('feature-toggles');
 const tu = require('../testUtils');
 
-describe('test/logger.js > ', () => {
+describe.only('test/logger.js > ', () => {
```
Comment (iamigo) [inline tests/logger/logger.js]: .only?

File: index.js [hunk 5]
```diff
@@ -84,17 +86,23 @@ function start(clusterProcessId = 0) { // eslint-disable-line max-statements
...
-  console.log('Started node cluster master');
+  logger.info('Started node cluster master');
 } // startMaster
 
+function initApp() {
+  initKafkaLoggingProducer().then(start).catch((err) => {
+    logger.error(err);
+  });
+}
+
```
Comment (jgraff2) [inline index.js]: same comment as other PR - I think the toggle check should be done here so it's clear we're only initializing the kafka producer when the toggle is enabled.

File: public/scripts/socket.io.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+const logger = require('../../logger');
```
Comment (jgraff2) [inline public/scripts/socket.io.js]: Not needed for this file.

File: migrations/20190603225918-no-paranoid.js [hunk 7]
```diff
@@ -16,6 +16,8 @@
 'use strict';
 const Promise = require('bluebird');
 const db = require('../db/index');
+const logger = require('../logger');
```
Comment (jgraff2) [inline migrations/20190603225918-no-paranoid.js]: Not needed for this file.

File: tests/api/v1/lenses/get.js [hunk 8]
```diff
@@ -15,6 +15,7 @@ const api = supertest(require('../../../../express').app);
 const constants = require('../../../../api/v1/constants');
 const tu = require('../../../testUtils');
 const u = require('./utils');
+const logger = require('../../../../logger');
```
Comment (jgraff2) [inline tests/api/v1/lenses/get.js]: Not needed in test files.

File: tests/cache/jobQueue/bulkUpsert.js [hunk 9]
```diff
@@ -140,7 +140,7 @@ describe('tests/cache/jobQueue/bulkUpsert.js, ' +
       //don't call done() yet, need to wait for data to be logged
     });
 
-    function testLogMessage(transport, level, msg, meta) {
```
Comment (jgraff2) [inline tests/cache/jobQueue/bulkUpsert.js]: Is this change related to the PR, or just miscellaneous cleanup?

File: view/perspective/app.js [hunk 10]
```diff
@@ -49,6 +49,7 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import PerspectiveController from './PerspectiveController';
 import { getValuesObject } from './utils';
+const logger = require('../../logger');
```
Comment (jgraff2) [inline view/perspective/app.js]: Not needed in this file.

File: worker/index.js [hunk 11]
```diff
@@ -17,7 +17,7 @@
  */
 'use strict'; // eslint-disable-line strict
 const featureToggles = require('feature-toggles');
-const logger = require('winston');
+const logger = require('../logger');
```
Comment (jgraff2) [inline worker/index.js]: Need to init the producer here as well, since this is an entry point for a separate process.

File: clock/index.js [hunk 12]
```diff
@@ -23,6 +23,7 @@
  * clockJobConfig object in config.js.
  */
 const conf = require('../config');
+const logger = require('../logger');
```
Comment (jgraff2) [inline clock/index.js]: Need to init the producer here as well, since this is an entry point for a separate process.

**Pull request closed without merging**